src/main/java/com/jcabi/http/wire/AbstractHeaderBasedCachingWire.java

```diff
@@ -93,9 +93,22 @@ public final Response send(
     ) throws IOException {
         final Response rsp;
         if (method.equals(Request.GET)) {
-            rsp = this.consultCache(
-                req, home, method, headers, content, connect, read
-            );
+            boolean requestHasCmch = false;
+            for (final Map.Entry<String, String> header : headers) {
+                if (header.getKey().equals(this.cmch)) {
+                    requestHasCmch = true;
+                    break;
+                }
+            }
+            if (requestHasCmch) {
+                rsp = this.origin.send(
```

Comment:
@amihaiemil we have code duplication between these lines and 112-116. Can you extract the above logic to a method and get rid of duplication by adding additional condition to ` if (method.equals(Request.GET)) {`?